aries_cloudagent/admin/server.py

```diff
@@ -725,24 +726,60 @@ def remove_webhook_target(self, target_url: str):
         if target_url in self.webhook_targets:
             del self.webhook_targets[target_url]
 
+    async def dispatch_subwallet_webhook(
+        self,
+        topic: str,
+        payload: dict,
+        webhook_url: str,
+        metadata: dict,
+        max_attempts: int = None,
+    ):
+        """Route a webhook associated with a subwallet using webhook_router."""
+        self.webhook_router(
+            topic=topic,
+            payload=payload,
+            endpoint=webhook_url,
+            metadata=metadata,
+        )
+
     async def send_webhook(self, profile: Profile, topic: str, payload: dict):
         """Add a webhook to the queue, to send to all registered targets."""
         wallet_id = profile.settings.get("wallet.id")
-
+        wallet_name = profile.settings.get("wallet.name")
+        wallet_dispatch_type = profile.settings.get("wallet.dispatch_type")
+        wallet_webhook_urls = profile.settings.get("wallet.webhook_urls")
+        base_wallet = wallet_id == wallet_name
+
+        # async with profile.session() as session:
+        #     wallet = await WalletRecord.retrieve_by_id(session, wallet_id)
+        #     base_wallet = wallet.wallet_id == wallet.wallet_name
+        #     wallet_dispatch_type = wallet.wallet_dispatch_type
+        #     wallet_webhook_urls = wallet.wallet_webhoook_urls
```

Comment:
The wallet record cannot be found in this case because the profile is in context of the subwallet. All wallet records are stored in the base wallet. This should work if you use `self.root_profile` instead of `profile`, but I think you can get all required settings from the settings dict